docs/TuxStores.md

```diff
@@ -0,0 +1,182 @@
+
+```
+    var listenerCallback = function () {
+      this.setState({ messages: MessageStore.all() });
+    };
+
+    store.addChangeListener(listenerCallback);
+```
+
+##### <a id="addChangeListener-EVENT_TYPE"></a>Parameter - `EVENT_TYPE` - type: STRING - optional [#](#addChangeListener-EVENT_TYPE)
+Event string to add a listener callback to. If no event is passed in than the listener will be added to the default `CHANGE` event.
+
+```
+    var listenerCallback = function () {
+      this.setState({ messages: MessageStore.all() });
+    };
+
+    var newEventType = 'NEW_EVENT_TYPE';
+    store.addChangeListener(listenerCallback, newEventType);
+```
+
+
+### <a id="store-removeChangeListener"></a>Method `store.removeChangeListener` - type: FUNCTION [#](#store-removeChangeListener)
+Removes the passed in listener callback from the passed in event string. If no event is specified than the default `CHANGE` event is used.
+
+```
+    store.removeChangeListener(callback, EVENT_TYPE);
+```
+
+##### <a id="removeChangeListener-callback"></a>Parameter - `callback` - type: FUNCTION - required [#](#removeChangeListener-callback)
+The callback function that will be removed from the stores set of listeners.
+
+```
+    var listenerCallback = function () {
+      this.setState({ messages: MessageStore.all() });
+    };
+
+    store.addChangeListener(listenerCallback);
+    store.removeChangeListener(listenerCallback);
+```
+
+##### <a id="removeChangeListener-EVENT_TYPE"></a>Parameter - `EVENT_TYPE` - type: STRING - optional [#](#removeChangeListener-EVENT_TYPE)
+Change event type string that will be used to remove the listener from the store. If no `EVENT_TYPE` is specified the default value of `CHANGE` will be used.
+
+```
+    var listenerCallback = function () {
+      this.setState({ messages: MessageStore.all() });
+    };
+
+    var newEventType = 'NEW_EVENT_TYPE';
+    store.addChangeListener(listenerCallback, newEventType);
+    store.removeChangeListener(listenerCallback, newEventType);
+```
+
+## <a id="TuxStores-Complete-Example"></a>TuxStores Complete Example [#](#TuxStores-Complete-Example)
+Taking everything we have learned lets look at a more realistic example of using TuxStores.
+
+```
+    var Stores = require('tux/Stores');
+
+    var messageStore = Stores.createStore({
+      _messages: [],
+
+      allMessages: function () {
+        return this._messages;
+      },
+
+      onGet: function (data) {
+        this._messages = data.messages;
+        this.emitChange();
+      },
+
+      onCreate: function (message) {
+        this._messages.push(message);
+        this.emitChange();
+      },
+
+      onDelete: function (message) {
+        this._messages.splice(message.id, 1);
+        this.emitChange();
+      }
+    });
+
+    console.assert(messageStore.hasOwnProperty('emitChange'));
+    console.assert(messageStore.hasOwnProperty('addChangeListener'));
+    console.assert(messageStore.hasOwnProperty('removeChangeListener'));
+```
```

Comment:
After the console.assert can we add a really simple object that adds a listener for the store?  It doesn't even have to be a view component, just something that can invoke `addChangeListener` and add a listener to invoke `allMessages` on change, and then later remove the listener.